File: exercicios/java/Exercicio03/src/main/java/org/example/Venda.java
package org.example;

public class Venda {
    private int codigo;
    private Cliente cliente;
    private String descricao;
    private PacoteViagem pacote;
    private int quantidade;

    public Venda(int codigo, Cliente cliente, String descricao, PacoteViagem pacote, int i) {
        this.codigo = codigo;
        this.cliente = cliente;
        this.descricao = descricao;
        this.pacote = pacote;
        this.quantidade = quantidade;
    }
    public int getCodigo() {
        return codigo;
    }
    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }
    public Cliente getCliente() {
        return cliente;
    }
    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }
    public String getDescricao() {
        return descricao;
    }
    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }
    public PacoteViagem getPacote() {
        return pacote;
    }
    public void setPacote(PacoteViagem pacote) {
        this.pacote = pacote;
    }
    public int getQuantidade() {
        return quantidade;
    }
    public void setQuantidade(int quantidade) {
        this.quantidade = quantidade;
    }
    public double precoTotal(){
        double preco = this.quantidade * PacoteViagem.getCustoUnitario();
        return preco;
    }
}

File: exercicios/python/Exercício 03 - Implementação de Classes e Objetos (Pacote de Viagem) (25_08)/venda.py
from pacote_viagem import PacoteViagem
from cliente import Cliente

class Venda:
    def __init__(self, codigo: str, cliente: Cliente, descricao:str, pacote: PacoteViagem, quantidade: int):
        self.__codigo = codigo
        self.__cliente = cliente
        self.__descricao = descricao
        self.__pacote = pacote
        self.__quantidade = quantidade
        
    @property
    def codigo(self):
        return self.__codigo
    
    @codigo.setter
    def codigo(self, codigo: str):
        self.__codigo = codigo
        
    @property
    def cliente(self):
        return self.__cliente
    
    @cliente.setter
    def cliente(self, cliente: Cliente):
        self.__cliente = cliente
    
    @property
    def descricao(self):
        return self.__descricao
    
    @descricao.setter
    def descricao(self, descricao: str):
        self.__descricao = descricao

    @property
    def pacote(self):
        return self.__pacote
    
    @pacote.setter
    def pacote(self, pacote: PacoteViagem):
        self.__pacote = pacote
        
    @property
    def quantidade(self):
        return self.__quantidade
    
    @quantidade.setter
    def quantidade(self, quantidade: int):
        self.__quantidade = quantidade

    def preco_total(self):
        return self.__quantidade * self.__pacote.custo_unitario
